### DISCUSSION TITLE
Image component causes unexpected behavior with Styled-Components

### DISCUSSION THREAD

**Discussion by BjoernRave** [vercel/next.js#18380] [Help] [answered]
When using the `Image` component together with `styled-components` like this:

```
import Image from 'next/image'
import styled from 'styled-components'

const StyledImage = styled(Image)`
...
`
```
the styles are not applied at the right place, because the image component is rendering a `div` around the img, but applying the styles to the `img` element

**Comment by merodiro** [comment]
I also noticed the same problem when using it with TailwindCSS

**Reply by sfonua10** [reply]
Yep, I saw the same problem with TailwindCSS

**Comment by jamesmosier** [comment, marked as answer]
I was able to get this to work just fine (using Next v10 and styled-components v5.2.0):

```js
import styled from 'styled-components';
import Image from 'next/image';

const StyledImage = styled(Image)`
  border-radius: 10px;
`;

export default function Home() {
  return (
    <StyledImage
      src="/computer.jpg"
      alt="Picture of a computer"
      width={500}
      height={500}
    />
  );
}
```

---

Is there something specific that is not working?

**Reply by BjoernRave** [reply]
positioning images is not working, since the image component is rendering additional divs, but applying the styles to the `img` :

```
const StyledImage = styled(Image)`
 position: absolute;
  left: 10%;
  top: 10%;
  width: 50%;
`
```

**Reply by BjoernRave** [reply]
wouldnt it make sense  to pass `className` and `style` to the outer `div` ?

**Reply by jamesmosier** [reply]
But then others would say "why are the styles not applied to the image itself?" 😄  

I would just create a wrapper div and style that accordingly for your positioning:

```js
const StyledImage = styled(Image)`
  border-radius: 10px;
`;

const Wrap = styled.div`
  position: absolute;
  left: 10%;
  top: 10%;
  width: 50%;
`;

export default function Home() {
  return (
    <div style={{ position: 'relative', height: '2000px', background: '#ccc' }}>
      <Wrap>
        <StyledImage
          src="/computer.jpg"
          alt="Picture of a computer"
          width={500}
          height={500}
        />
      </Wrap>
    </div>
  );
}
```

**Reply by BjoernRave** [reply]
Yes, that's possible I guess, but it makes working with the image component a little more cumbersome. I guess there is no other option though, right?

**Reply by jamesmosier** [reply]
Correct, as of right now there is no other option.

**Reply by BjoernRave** [reply]
maybe there could be two additional props: `wrapperClassName` and `wrapperStyle` or something along those lines, so you can pass styles/classNames to either the `img` tag or the outer `div`

**Reply by fmacedoo** [reply]
I am facing some issues like that as well. I wish to put an image layout=fill into a flex row, and I would like that row to grow as the image size. I was only able to accomplish that setting some `!important` **¯\_(ツ)_/¯** overriding the inlines `displays: relative` at image component.

**Reply by tigerabrodi** [reply]
I was just looking around, thanks first and foremost to all inputs here :heart: 

FYI Regards the alt text, it doesn't need to say that it is a picture or image, the screen reader will voice either "graphic" or "image" when voicing an image's alt, hence it is redundant :+1:

**Comment by acandael** [comment]
having the same issue here:

    export const SubSectionImage = styled(Image)`
    margin-bottom: 48px;
    `


but the SubSectionImage component is not picking up de defined style.

**Reply by ezeikel** [reply]
Wrapping it with another component until they fix this

e.g.

```
import styled from "styled-components";
import Image from "next/image";

const Wrapper = styled.div`
  display: flex;
`;

const NextJSImageWrapper = ({ className, ...props }) => (
  <Wrapper className={className}>
    <Image {...props} />
  </Wrapper>
);

export default NextJSImageWrapper;
````

**Comment by natemwilson** [comment]
This issue is preventing me from adding a border to an image. I want to do something like this:
```
const ProfilePicture  = styled(Image)`
  border-radius: 9999px;
  border: 3px solid black;
`
```

but Next.js automatically adds all these styles:
```
position: absolute; inset: 0px; box-sizing: border-box; padding: 0px; border: none; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;
```
one of which is `border: none` which undoes my border because of it's higher specificity! Any idea how I can get around this?

**Reply by jlasata** [reply]
I had the same issue - was able to override w/ !important

```
const ImageContainer = Styled(Image)`
  border: 1px solid green !important;
  border-radius: 5px;
`;
```

**Comment by JulinoSD** [comment]
I'm newby, and i'm triying to adding image as background-image on my project, i did all sort of think that i needed to do (like others are doing), and i can't get nothing, please help!
<pre>
import styled from 'styled-components'

export const Landing = styled.section`
    // background: red;
    background-image: url('../../public/images/fundo.png')
    console.log(background-image)
    width: 100vw;
    height: 100vh;
</pre>
Here the result:
![image](https://user-images.githubusercontent.com/51002858/151464062-2d6a72f4-4e0a-4c83-8f99-0ead007cd489.png)

**Reply by JulinoSD** [reply]
ignore the `console.log()` lol

**Comment by iamcrisb** [comment]
This issue still persists. Next.js is creating additional elements and adds inline styles on them. Totally unusable in this context.

**Comment by alfiejfs** [comment]
Just commenting to give this issue more visibility - would be a very nice thing to consider :)

**Comment by bernardoenock** [comment]
I did what I needed with background-image
Next brings us so many things... they could fix this...
